llmnr.c

```diff
@@ -40,38 +40,74 @@
 
 static bool llmnr_ipv6 = false;
 /* Host name in DNS name format (length octet + name + 0 byte) */
-static char llmnr_hostname[LLMNR_LABEL_MAX_SIZE + 2];
+#define LLMNR_LABEL_LEN (LLMNR_LABEL_MAX_SIZE + 2)
+static char** llmnr_hostnames = NULL;
+size_t llmnr_hostname_count = 0;
+
+static void set_hostname(int entry_index, const char *hostname)
+{
+	char* entry_llmnr_hostname = NULL;
+
+	llmnr_hostnames[entry_index] = xzalloc(LLMNR_LABEL_LEN);
+	entry_llmnr_hostname = llmnr_hostnames[entry_index];
+
+	entry_llmnr_hostname[0] = (uint8_t)strlen(hostname);
+	strncpy(&entry_llmnr_hostname[1], hostname, LLMNR_LABEL_MAX_SIZE);
+	entry_llmnr_hostname[LLMNR_LABEL_MAX_SIZE + 1] = '\0';
+}
 
 void llmnr_set_hostname(const char *hostname)
 {
-	llmnr_hostname[0] = strlen(hostname);
-	strncpy(&llmnr_hostname[1], hostname, LLMNR_LABEL_MAX_SIZE);
-	llmnr_hostname[LLMNR_LABEL_MAX_SIZE + 1] = '\0';
+	set_hostname(0, hostname);
 }
 
-void llmnr_init(const char *hostname, bool ipv6)
+void llmnr_init(const char *hostnames[], size_t hostname_count, bool ipv6)
 {
-	llmnr_set_hostname(hostname);
+	size_t i;
+	llmnr_hostname_count = hostname_count;
+	llmnr_hostnames = xzalloc(hostname_count);
+	for (i = 0; i < hostname_count; ++i) {
+		set_hostname(i, hostnames[i]);
+	}
 	llmnr_ipv6 = ipv6;
 }
 
-static bool llmnr_name_matches(const uint8_t *query)
+void llmnr_release(void)
+{
+	size_t i;
+	for(i = 0; i < llmnr_hostname_count; ++i) {
```

Comment:
Add an empty line after variable declaration and add a space after the `for` keyword:
```suggestion
	size_t i;

	for (i = 0; i < llmnr_hostname_count; ++i) {
```